modules/database/funcs.go

```diff
@@ -54,23 +54,25 @@ func NewDatabase(
 func (d *Database) setDatabaseHostname(
 	ctx context.Context,
 	h *helper.Helper,
+	labels map[string]string,
 ) error {
 
-	selector := map[string]string{
-		"app": "mariadb",
-	}
+	labels["app"] = "mariadb"
```

Comment:
I'm wondering if line 60 is actually need it as i would have guess the cr would have been enough

i assume ewe have callers that wont just pass the cr name as is being default in line 126 below

Reply:
Probably L126 is enough now. But I would like to keep the "app" filtering now. We can change this later easily. 

Currently the MariaDB CR  creation will result in the following Service instance:

```
$ oc describe service/openstack
Name:              openstack
Namespace:         openstack
Labels:            app=mariadb
                   cr=mariadb-openstack
                   owner=mariadb-operator
Annotations:       <none>
Selector:          app=mariadb
...
```
When an external service is defined via https://www.sobyte.net/post/2022-04/kubernetes-ext-service/ we might not have "cr" label (we could add them manually but it would not be true as there would be no CR behind that external DB). So we might want to filter for "app" still in that case and maybe use the Service.name filtering to select the nova-cell1 DB Service instance.

Reply:
ack that makes sense.

i was not pushing to drop this by the way just wanted to make sure i understood why you had boot when the cr would be enough in this specific case

lets keep it as you have it to maintain the same behavior as previously